lib/linter/apply-disable-directives.js

```diff
@@ -258,17 +333,45 @@ function applyDirectives(options) {
     const unusedDisableDirectivesToReport = options.directives
         .filter(directive => directive.type === "disable" && !usedDisableDirectives.has(directive));
 
-    const processed = processUnusedDisableDirectives(unusedDisableDirectivesToReport);
 
-    const unusedDisableDirectives = processed
+    const unusedEnableDirectivesToReport = new Set(
+        options.directives.filter(directive => directive.unprocessedDirective.type === "enable")
+    );
+
+    if (
+
+        /*
+         * If directives has the eslint-enable directive,
+         * check whether the eslint-enable comment is used.
+         */
+        unusedEnableDirectivesToReport.size > 0
+    ) {
+        for (const directive of collectUsedEnableDirectives(options.directives)) {
+            unusedEnableDirectivesToReport.delete(directive);
+        }
+    }
+
+    const processed = processUnusedDirectives(unusedDisableDirectivesToReport)
+        .concat(processUnusedDirectives(unusedEnableDirectivesToReport));
+
+    const unusedDirectives = processed
         .map(({ description, fix, unprocessedDirective }) => {
             const { parentComment, type, line, column } = unprocessedDirective;
 
+            let message;
+
+            if (type === "enable") {
+                message = description
+                    ? `Unused eslint-enable directive (no suppressed for ${description}).`
+                    : "Unused eslint-enable directive (no suppressed).";
```

Comment:
"Suppressed" relates more to lint problems that were found but suppressed, while this feature is only based on matching with disable directives.

Maybe something like this?

```suggestion
                    ? `Unused eslint-enable directive (no matching eslint-disable directives were found for ${description}).`
                    : "Unused eslint-enable directive (no matching eslint-disable directives were found).";
```